broker/broker.go

```diff
@@ -687,6 +672,43 @@ func (b *CdnServiceBroker) checkDomain(logger lager.Logger, domainsConcat, orgGU
 	return nil
 }
 
```

Comment:
Would it be better to ask for the list domains owned by an org and checking for the presence in that list, rather than making 2 API calls?

Reply:
I don't think we'd be able to differentiate between non-existent domains and ones owned by someone else then. Also we'd have to somehow include domains that are _shared with_ our org.

Reply:
I'm not sure what you mean by  "non-existent domains" here. 

[The `/v3/organizations/:guid/domains`](https://v3-apidocs.cloudfoundry.org/version/3.97.0/index.html#list-domains-for-an-organization) endpoint will show you the domains owned by that particular domain. It remains to be seen if that will include private shared domains.

Reply:
This way manages to give a different error message depending on whether the cf domain is owned by a different org or if the cf domain just hasn't been created at all.

Referring to a list of domains just owned by the current org would mean we'd have to do a follow-up request to find out if the domain exists in cf at all if we still wanted to differentiate the error messages.

Reply:
Ok, I think i understand it better now. We want two distinct code paths for the sake of better errors messages:

1. You didn't create the domain
2. You don't own the domain

Reviewing it in that light, I'll approve it.